File: EarthQuake/app/src/main/java 2/com/example/earthquake/MainActivity.java
package com.example.earthquake;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ArrayList<EarthQuake> words = QueryUtility.extractEarthquakes();

        EarthQuakeAdapter adapter = new EarthQuakeAdapter(this  , words);

        ListView lView = (ListView) findViewById(R.id.list);

        lView.setAdapter(adapter);

        lView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                EarthQuake currentEQ = adapter.getItem(position);
                Uri eQUri = Uri.parse(currentEQ.getmURL());

                Intent i = new Intent(Intent.ACTION_VIEW , eQUri);
                startActivity(i);
            }
        });
    }
}

File: BroadcastAndroid/app/src/main/java/com/example/broadcastandroid/MainActivity.kt
package com.example.broadcastandroid

import android.content.Intent
import android.content.IntentFilter
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {

    private lateinit var receiver: AirPlaneModeReceiver

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        receiver = AirPlaneModeReceiver()
        registerReceiver(receiver , IntentFilter(Intent.ACTION_AIRPLANE_MODE_CHANGED))
    }

    override fun onStop() {
        super.onStop()
        unregisterReceiver(receiver)
    }

}